source/type/type.h

```diff
@@ -722,9 +721,15 @@ struct VehicleJourney: public Header, Nameable, hasVehicleProperties, HasMessage
     int16_t utc_to_local_offset = 0; //in seconds
 
     RTLevel realtime_level = RTLevel::Theoric;
-    ValidityPattern* adapted_validity_pattern = nullptr; //REMOVE
+
+    // validity pattern for all RTLevel
+    flat_enum_map<RTLevel, ValidityPattern*> validity_patterns = {{{nullptr, nullptr, nullptr}}};
+
     std::vector<VehicleJourney*> adapted_vehicle_journey_list; //REMOVE
-    VehicleJourney* theoric_vehicle_journey = nullptr; //REMOVE
+    VehicleJourney* theoric_vehicle_journey = nullptr;
+
+    ValidityPattern* adapted_validity_pattern() const { return validity_patterns[RTLevel::Adapted]; }
+    ValidityPattern* validity_pattern() const { return validity_patterns[RTLevel::Theoric]; }
```

Comment:
Not fan of implicit "not saying it's adapted or realtime so it's theoric"... I'd rather use `theoric_validity_pattern`
It might be refactored soon though, but still, it might ease the refactor.